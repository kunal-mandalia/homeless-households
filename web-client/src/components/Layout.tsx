import * as React from 'react';
import styled from 'styled-components';
import { COLORS } from '../constants';

interface IProps {
  sidebar: JSX.Element,
  tiles: JSX.Element[],
}

const LayoutWrapper = styled.div`
  display: flex;
  flex-direction: row;
  flex-wrap: wrap-reverse;
  min-height: 100vh;
`

const SidebarWrapper = styled.div`
  background-color: ${COLORS.MAROON};
  flex-basis: 385px;
  flex-grow: 0;
`

const TilesWrapper = styled.div`
  display: flex;
  flex: 1;
  flex-grow: 8;
  flex-wrap: wrap;
  border: solid 4px ${COLORS.MAROON};
`

const Tile = styled.div`
  flex: 1;
  flex-direction: column;
  flex-basis: 300px;
  border: solid 4px ${COLORS.MAROON};
  min-height: 200px;
  background-color: ${COLORS.MAROON_LIGHT};
`


export const Layout = ({ sidebar, tiles } : IProps) => {
  return <LayoutWrapper>
    <SidebarWrapper>{sidebar}</SidebarWrapper>
    <TilesWrapper>
      {tiles.map((tile, id) => <Tile key={id} data-count={tiles.length}>{tile}</Tile>)}
    </TilesWrapper>
  </LayoutWrapper>
}
