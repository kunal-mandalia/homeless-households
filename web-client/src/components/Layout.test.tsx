import { shallow } from 'enzyme';
import * as React from 'react';
import { Layout } from './Layout';

describe('<Layout .../>', () => {
  it('should render one sidebar', () => {
    // arrange
    const sidebar = <div data-testid='sidebar'>Sidebar</div>
    // act
    const wrapper = shallow(<Layout sidebar={sidebar} tiles={[]} />)
    // assert
    expect(wrapper.find(`[data-testid="sidebar"]`)).toHaveLength(1)
  })

  it('should render many tiles', () => {
    // arrange
    const tiles = [
      <div key='tile-age' data-testid='tile-age'/>,
      <div key='tile-need' data-testid='tile-need'/>,
    ]
    // act
    const wrapper = shallow(<Layout sidebar={<div />} tiles={tiles} />)
    // assert
    expect(wrapper.find(`[data-testid="tile-age"]`)).toHaveLength(1)
    expect(wrapper.find(`[data-testid="tile-need"]`)).toHaveLength(1)
  })

  it('should match snapshot', () => {
    // arrange
    const sidebar = <div data-testid='sidebar'>Sidebar</div>
    const tiles = [
      <div key='tile-age' data-testid='tile-age'/>,
      <div key='tile-need' data-testid='tile-need'/>,
    ]
    // act
    const wrapper = shallow(<Layout sidebar={sidebar} tiles={tiles} />)
    // assert
    expect(wrapper).toMatchSnapshot()
  })
})
