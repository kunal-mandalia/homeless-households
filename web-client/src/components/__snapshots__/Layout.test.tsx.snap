// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout .../> should match snapshot 1`] = `
<styled.div>
  <styled.div>
    <div
      data-testid="sidebar"
    >
      Sidebar
    </div>
  </styled.div>
  <styled.div>
    <styled.div
      data-count={2}
      key="0"
    >
      <div
        data-testid="tile-age"
        key="tile-age"
      />
    </styled.div>
    <styled.div
      data-count={2}
      key="1"
    >
      <div
        data-testid="tile-need"
        key="tile-need"
      />
    </styled.div>
  </styled.div>
</styled.div>
`;
